bench-tps/src/cli.rs

```diff
@@ -496,22 +529,133 @@ pub fn extract_args(matches: &ArgMatches) -> Config {
             .value_of("instruction_padding_program_id")
             .map(|target_str| target_str.parse().unwrap())
             .unwrap_or_else(|| FromOtherSolana::from(spl_instruction_padding::ID));
+        let parsed_data_size;
+        assign_or_return_err!(
+            parsed_data_size,
+            data_size.to_string().parse::<u32>(),
+            "Can't parse padded instruction data size"
+        );
         args.instruction_padding_config = Some(InstructionPaddingConfig {
             program_id,
-            data_size: data_size
-                .to_string()
-                .parse()
-                .expect("Can't parse padded instruction data size"),
+            data_size: parsed_data_size,
         });
     }
 
     if let Some(num_conflict_groups) = matches.value_of("num_conflict_groups") {
-        args.num_conflict_groups = Some(
-            num_conflict_groups
-                .parse()
-                .expect("Can't parse conflict groups"),
+        let parsed_num_conflict_groups;
+        assign_or_return_err!(
+            parsed_num_conflict_groups,
+            num_conflict_groups.parse(),
+            "Can't parse conflict groups"
         );
+        args.num_conflict_groups = Some(parsed_num_conflict_groups);
     }
 
-    args
+    Ok(args)
+}
+
+mod tests {
+    use {
+        super::*,
+        solana_sdk::signature::{read_keypair_file, write_keypair_file, Keypair, Signer},
+    };
+
+    fn make_tmp_path(name: &str) -> String {
+        let out_dir = std::env::var("FARF_DIR").unwrap_or_else(|_| "farf".to_string());
+        let keypair = Keypair::new();
+
+        let path = format!("{}/tmp/{}-{}", out_dir, name, keypair.pubkey());
+
+        // whack any possible collision
+        let _ignored = std::fs::remove_dir_all(&path);
```

Comment:
*&lt;minor-style&gt;*
When a result is ignored, it is normally spelled as just
```rust
let _ = f();
```
A name is added after the underscore when you want to explicitly name the thing you are ignoring.  Saying `_ignored` is redundant.
*&lt;/minor-style&gt;*

Unless you actually expect the failure to be acceptable, it is better to use
```rust
    std::fs::remove_dir_all(&path)
        .expect("Removal of all the temp directory is successful");
```

Wouldn't it create a flaky test if it indeed overlaps?
In which case, maybe, instead of removing the path, it should panic?

Also, there are some nice libraries for creating temporary directories.
`tempfile` seems to be pretty popular: https://crates.io/crates/tempfile
It will create a temporary directory that should not conflict with anything, and will also delete it, when it goes out of scope.

I do not know what `FARF_DIR` is.  So `tempfile` may or may not be what you want.

Reply:
Hm, probably a good idea to move to this crate

Reply:
I've added, `tempfile` is difficult to use because I need to have a name of the file to pass to cli. Experience with tempdir is smoother